packages/wrangler/src/dev.ts

```diff
@@ -560,216 +700,154 @@ export async function startDev(args: StartDevOptions) {
-							if (hotkeysDisplayed) {
-								unregisterHotKeys = registerDevHotKeys(devEnv, args);
+				devEnv.runtimes.forEach((runtime) => {
+					runtime.on(
+						"reloadComplete",
+						async (reloadEvent: ReloadCompleteEvent) => {
+							if (!reloadEvent.config.dev?.remote) {
+								assert(devEnv !== undefined && !Array.isArray(devEnv));
+								const { url } = await devEnv.proxy.ready.promise;
+
+								await maybeRegisterLocalWorker(
+									url,
+									reloadEvent.config.name,
+									reloadEvent.proxyData.internalDurableObjects,
+									reloadEvent.proxyData.entrypointAddresses
+								);
 							}
 						}
-						return {
-							accountId,
-							apiToken: requireApiToken(),
-						};
-					},
-					remote: !args.forceLocal && args.remote,
-					server: {
-						hostname: args.ip,
-						port: args.port,
-						secure:
-							args.localProtocol === undefined
-								? undefined
-								: args.localProtocol === "https",
-						httpsCertPath: args.httpsCertPath,
-						httpsKeyPath: args.httpsKeyPath,
-					},
-					inspector: {
-						port: args.inspectorPort,
-					},
-					origin: {
-						hostname: args.host ?? args.localUpstream,
-						secure:
-							args.upstreamProtocol === undefined
-								? undefined
-								: args.upstreamProtocol === "https",
-					},
-					persist: args.persistTo,
-					liveReload: args.liveReload,
-					testScheduled: args.testScheduled,
-					logLevel: args.logLevel,
-					registry: devEnv.config.latestConfig?.dev.registry,
-					bindVectorizeToProd: args.experimentalVectorizeBindToProd,
-				},
-				legacy: {
-					site: (configParam) => {
-						const legacyAssetPaths = getResolvedLegacyAssetPaths(
-							args,
-							configParam
-						);
-						return Boolean(args.site || configParam.site) && legacyAssetPaths
-							? {
-									bucket: path.join(
-										legacyAssetPaths.baseDirectory,
-										legacyAssetPaths?.assetDirectory
-									),
-									include: legacyAssetPaths.includePatterns,
-									exclude: legacyAssetPaths.excludePatterns,
-								}
-							: undefined;
-					},
-					legacyAssets: (configParam) =>
-						args.legacyAssets ?? configParam.legacy_assets,
-					enableServiceEnvironments: !(args.legacyEnv ?? true),
-				},
-				assets: args.assets,
-			} satisfies StartDevWorkerInput,
-			true
-		);
+					);
+				});
+			}
+
+			if (isInteractive() && args.showInteractiveDevSession !== false) {
+				unregisterHotKeys = registerDevHotKeys(devEnv, args);
+			}
+
+			await setupDevEnv(devEnv, configPath, authHook, args);
+		}
 
 		void metrics.sendMetricsEvent(
 			"run dev",
 			{
 				local: !args.remote,
-				usesTypeScript: /\.tsx?$/.test(
-					devEnv.config.latestConfig?.entrypoint as string
-				),
+				// usesTypeScript: /\.tsx?$/.test(
+				// 	devEnv.config.latestConfig?.entrypoint as string
+				// ),
 			},
 			{
-				sendMetrics: devEnv.config.latestConfig?.sendMetrics,
+				// sendMetrics: devEnv.config.latestConfig?.sendMetrics,
```

Comment:
bit concerned about ignoring send_metrics. should we remove this metrics call?

Reply:
Removed in e14865fef